### PR TITLE
[TS migration] Migrate IOURequestStepTag/Participants/Merchant

### PR DESCRIPTION
**Pull request by ruben-rebelo**
### Details
[TS migration] Migrate IOURequestStepTag/Participants/Merchant

### Fixed Issues
$ https://github.com/Expensify/App/issues/38916
PROPOSAL:

### Tests
Workspace must have tags enabled
- Request money to a workspace (Participant list works)
- Add tag, Merchant to the request (Tag and merchant is added)

- [ ] Verify that no errors appear in the JS console

### Offline tests
N/A

### QA Steps
Same as in Tests section

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/39693995/1e919a17-6f74-4962-b1e9-b9fbce6d56a3

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/39693995/581bf803-e298-4b43-b19e-4df2e48e8670

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/39693995/0b045c13-2e06-4a1e-8355-ddb095fd8daa

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/39693995/eb6e4b5a-9a58-4fa7-a22f-27ef9a4edd03

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/39693995/c759ce5b-63a5-4603-bf4c-ef8915185077

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/39693995/ff049015-7acf-4fca-abdb-061ea8d43647

</details>

### CODE REVIEW

**Comment by kubabutkiewicz** [comment]
@ruben-rebelo can you check lint and typecheck steps?

**Comment by fabioh8010** [comment]
@ruben-rebelo Why is this PR on hold for #38990? Can we move with this one without having to wait?

**Comment by fabioh8010** [comment]
Also we have conflicts!

**Comment by fabioh8010** [comment]
@ruben-rebelo #38990 was merged! 🎉

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (kubabutkiewicz) [inline src/pages/iou/request/step/IOURequestStepMerchant.tsx:21]: Maybe lets try to not use this type

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (fabioh8010) [inline src/pages/iou/request/step/IOURequestStepMerchant.tsx:85]:
```suggestion
            if (isMerchantRequired && (value.moneyRequestMerchant ?? '').length === 0) {
```

it will never be less than 0

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (fabioh8010) [inline src/pages/iou/request/step/IOURequestStepMerchant.tsx:97]: Let's remove everything

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (fabioh8010) [inline src/pages/iou/request/step/IOURequestStepMerchant.tsx:117]:
```suggestion
            IOU.updateMoneyRequestMerchant(transactionID, reportID, newMerchant || CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT, policy, policyTags, policyCategories);
```

We will introduce a regression if we use `??` here

File: src/pages/iou/request/step/IOURequestStepParticipants.tsx
Comment (fabioh8010) [inline src/pages/iou/request/step/IOURequestStepParticipants.tsx:38]:
```suggestion
type IOURef = ValueOf<typeof CONST.IOU.TYPE> | null;
```

**Review by fabioh8010 — COMMENTED** [review]
will continue review later

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (VickyStash) [inline src/pages/iou/request/step/IOURequestStepMerchant.tsx:83]: Since this function relates to form validation, maybe we can use `FormOnyxValues` and `FormInputErrors` types as in other places (there are a lot of examples through the app)

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (VickyStash) [reply src/pages/iou/request/step/IOURequestStepMerchant.tsx:85]: Since `value.moneyRequestMerchant` is string
```suggestion
            if (isMerchantRequired && !value.moneyRequestMerchant) {
```

File: src/pages/iou/request/step/IOURequestStepParticipants.tsx
Comment (VickyStash) [inline src/pages/iou/request/step/IOURequestStepParticipants.tsx:77]: Make sure this update doesn't cause regression

File: src/pages/iou/request/step/IOURequestStepParticipants.tsx
Comment (VickyStash) [inline src/pages/iou/request/step/IOURequestStepParticipants.tsx:94]: Maybe we can create a type for `ValueOf<typeof CONST.IOU.TYPE>`?

File: src/pages/iou/request/step/IOURequestStepParticipants.tsx
Comment (VickyStash) [inline src/pages/iou/request/step/IOURequestStepParticipants.tsx:124]:
```suggestion
            selectedReportID.current = val[0]?.reportID ?? reportID;
```

File: src/pages/iou/request/step/IOURequestStepTag.tsx
Comment (VickyStash) [inline src/pages/iou/request/step/IOURequestStepTag.tsx:32]: Do we still need it?

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (VickyStash) [inline src/pages/iou/request/step/IOURequestStepMerchant.tsx:95]:
```suggestion
    const updateMerchant = (value: FormOnyxValues<typeof ONYXKEYS.FORMS.MONEY_REQUEST_MERCHANT_FORM>) => {
```

File: src/pages/iou/request/step/IOURequestStepParticipants.tsx
Comment (fabioh8010) [reply src/pages/iou/request/step/IOURequestStepParticipants.tsx:79]: +1

**Comment by fabioh8010** [comment]
@ruben-rebelo we have lint errors now

**Review by kubabutkiewicz — APPROVED** [review]
lgtm!

**Review by blazejkustra — APPROVED** [review]

**Comment by ruben-rebelo** [comment]
@blazejkustra You are right, Updated!

File: src/libs/TransactionUtils.ts
Comment (mananjadhav) [inline src/libs/TransactionUtils.ts:52]: Could we have `lodashHas(transaction, 'iouRequestType')` replaced it with `transaction?.iouRequestType`?

File: src/libs/actions/IOU.ts
Comment (mananjadhav) [inline src/libs/actions/IOU.ts:317]: Are we sure making `tab?` won't have any impact?

**Review by mananjadhav — COMMENTED** [review]
Did partial review, left to review the tsx. Will need time to compare with js. Will get to it in next few hours.

File: src/libs/actions/IOU.ts
Comment (ruben-rebelo) [reply src/libs/actions/IOU.ts:317]: In this specific case we had to make it optional as it was never passed a tab prop.
I've tested the other calls of the functions and everything was working as expected.

File: src/libs/TransactionUtils.ts
Comment (ruben-rebelo) [reply src/libs/TransactionUtils.ts:52]: We can, but this was not done as this is not part of this issue.
Should we change it? wdyt?

**Comment by ruben-rebelo** [comment]
@mananjadhav I've updated the PR based on the withFullTransactionOrNotFound HOC that got merged.
Please have a look!

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (mananjadhav) [inline src/pages/iou/request/step/IOURequestStepMerchant.tsx:103]: any specific reason we want to disable this lint?

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/3069065/04e6e9b9-9a5b-47fb-92bc-fcd5d88f911c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/3069065/5ccb1e97-c403-442f-aff5-a80a51aeaf3a

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/3069065/01e31c5c-3386-46bf-91ae-0fb3f00180b7

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/3069065/207e6a1b-f0c3-4d64-bc44-e69ac4cc68a3

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/3069065/a615add6-9c28-434e-aeb4-2990f624ad89

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/3069065/882e676e-a367-4045-86a7-2e72451a52d3

</details>

**Review by mananjadhav — APPROVED** [review]

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (ruben-rebelo) [reply src/pages/iou/request/step/IOURequestStepMerchant.tsx]: This is a left-over, removed it, thanks for pointing it out!

**Comment by mananjadhav** [comment]
@ruben-rebelo I see some Typescript errors after the latest main merge. Can you take a long?

**Comment by fabioh8010** [comment]
@ruben-rebelo Please do the following changes to fix the TS errors.

```diff
diff --git a/src/components/ScreenWrapper.tsx b/src/components/ScreenWrapper.tsx
index b78e274371..e7d311e658 100644
--- a/src/components/ScreenWrapper.tsx
+++ b/src/components/ScreenWrapper.tsx
@@ -25,7 +25,7 @@ import SafeAreaConsumer from './SafeAreaConsumer';
 import TestToolsModal from './TestToolsModal';
 import withNavigationFallback from './withNavigationFallback';
 
-type ChildrenProps = {
+type ScreenWrapperChildrenProps = {
     insets: EdgeInsets;
     safeAreaPaddingBottomStyle?: {
         paddingBottom?: DimensionValue;
@@ -35,7 +35,7 @@ type ChildrenProps = {
 
 type ScreenWrapperProps = {
     /** Returns a function as a child to pass insets to or a node to render without insets */
-    children: ReactNode | React.FC<ChildrenProps>;
+    children: ReactNode | React.FC<ScreenWrapperChildrenProps>;
 
     /** A unique ID to find the screen wrapper in tests */
     testID: string;
@@ -281,3 +281,5 @@ function ScreenWrapper(
 ScreenWrapper.displayName = 'ScreenWrapper';
 
 export default withNavigationFallback(forwardRef(ScreenWrapper));
+
+export type {ScreenWrapperChildrenProps};
diff --git a/src/pages/iou/request/step/StepScreenWrapper.tsx b/src/pages/iou/request/step/StepScreenWrapper.tsx
index e64f2792d2..93285d359d 100644
--- a/src/pages/iou/request/step/StepScreenWrapper.tsx
+++ b/src/pages/iou/request/step/StepScreenWrapper.tsx
@@ -1,8 +1,9 @@
+import type {ReactNode} from 'react';
 import React from 'react';
-import type {PropsWithChildren} from 'react';
 import {View} from 'react-native';
 import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import type {ScreenWrapperChildrenProps} from '@components/ScreenWrapper';
 import ScreenWrapper from '@components/ScreenWrapper';
 import useThemeStyles from '@hooks/useThemeStyles';
 import * as DeviceCapabilities from '@libs/DeviceCapabilities';
@@ -29,6 +30,9 @@ type StepScreenWrapperProps = {
 
     /** Whether or not to include safe area padding */
     includeSafeAreaPaddingBottom?: boolean;
+
+    /** Returns a function as a child to pass insets to or a node to render without insets */
+    children: ReactNode | React.FC<ScreenWrapperChildrenProps>;
 };
 
 function StepScreenWrapper({
@@ -40,11 +44,11 @@ function StepScreenWrapper({
     shouldShowWrapper,
     shouldShowNotFoundPage,
     includeSafeAreaPaddingBottom,
-}: PropsWithChildren<StepScreenWrapperProps>) {
+}: StepScreenWrapperProps) {
     const styles = useThemeStyles();
 
     if (!shouldShowWrapper) {
-        return <FullPageNotFoundView shouldShow={shouldShowNotFoundPage}>{children}</FullPageNotFoundView>;
+        return <FullPageNotFoundView shouldShow={shouldShowNotFoundPage}>{children as ReactNode}</FullPageNotFoundView>;
     }
 
     return (

```

**Review by puneetlath — APPROVED** [review]

File: src/pages/iou/request/step/IOURequestStepMerchant.tsx
Comment (ikevin127) [inline src/pages/iou/request/step/IOURequestStepMerchant.tsx:100]: This empty string fallback caused issue https://github.com/Expensify/App/issues/41939 where if we set the merchant to something else first, for example "test", then the transaction draft merchant becomes "test".
`merchant: 'test'`
Then if we clear it, becomes empty string.
`merchant: ''`

If we submit an expense with an empty string of merchant, the BE response will return `Expense` as the merchant.

**Note:** The fallback should've been `newMerchant || CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT` like it is below for `updateMoneyRequestMerchant` function.

**Pull request merged**
